### PR TITLE
[doc] Add language re. current usage of devmode

### LINKED ISSUE #14063: [tlul, rtl] Use of devmode_i
On reviewing some OTBN RTL I realised the `devmode_i` input of the autogenerated `otbn_reg_top` is hard-wired to 1.

The documentation for register tool (here: https://docs.opentitan.org/doc/rm/register_tool/#error-responses) explains that the `devmode_i` input is meant to indicate if the device is in development or production mode. The idea being in production mode, for security critical blocks, rather than returning error messages on invalid accesses writes are silently ignored and reads return dummy data.

Looking at other blocks it appears all of them have `devmode_i` hard-wired to 1.

Is this an oversight? Should we have `devmode_i` suitably connected up to the life cycle controller or is this an old proposed feature we decided not to do?

### PR DESCRIPTION
**Pull request by tjaychen**
- fixes #14063

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: doc/rm/register_tool/index.md [hunk 1]
```diff
@@ -303,6 +303,11 @@ This signal indicates whether the whole SOC device is in development or producti
 For security peripherals in production mode, it is desired to **not** send an error response, so write misses silently fail, and read misses silently fail, but return either random data (TBD) or all `1`s for response data (i.e. `0xFFFFFFFF` for a 32b register).
 For non-security peripherals, or when in development mode (`devmode_i == 1`) these writes and reads to undefined addresses will return with TL-UL error response.
 
+#### Current Usage of devmode_i
```
Comment (msfschaffner) [inline doc/rm/register_tool/index.md:306]: This sounds good, just two comments:
1) only `devmode_i == 1` returns an error, and I think we set it to 1 everywhere (see line 304 above).
2) we may want to move this paragraph further down, right before `## Register definitions per type` since the following error descriptions `Other error responses (always sent, regardless of ...` seem unrelated to the current usage of devmode.

**Review by msfschaffner — APPROVED** [review]

File: doc/rm/register_tool/index.md (hunk 1 above)
Comment (tjaychen) [reply doc/rm/register_tool/index.md:306]: oops, reversed the polarity, let me fix that.
